src/CSModule.sol

```diff
@@ -1388,6 +1447,22 @@ contract CSModule is
         }
     }
 
+    function _isEnoughStrikesToEject(
+        uint256 strikesLifetime,
+        uint256 ejectThreshold,
+        uint256[] calldata strikesData
+    ) internal view returns (bool) {
+        uint256 strikesDataLength = strikesData.length;
+        if (strikesDataLength != strikesLifetime) revert InvalidStrikesData();
```

Comment:
what stops me of sending a proof for a stale key, such as:
- these are not the last X strikes but some from the past frames
- key has been withdrawn or exited already

Reply:
Oracle is in charge of setting the root with actual and valid strikes

Reply:
Note about withdrawn is valid